perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run YCSB benchmark against AWS DynamoDB.
```

Comment:
In our benchmarks, we usuallly try to avoid cloud providers. However, I see this is done in the cloud spanner YCSB benchmark.

A better example is the cloud_redis_ycsb_benchmark which spins up a "cloud redis" instance that may be either Google Memorystore for Redis or AWS Elasticache for Redis. In the providers directories, we then build up the cloud redis database. 

For example, see https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/aws/aws_elasticache_redis.py where we create an AWS elasticache for redis instance.

Reply:
yes, i based this on the gcp implementations, but if needed i could look into downloading a local copy of dynamodb and running against that, might defeat the purpose though?

Reply:
Ah, I meant we try to avoid access cloud providers directly from the benchmark. Instead, we'd add the code to configure dynamodb in providers/aws/aws_dynamodb.py. There we can inherit from a base class that the benchmark can access.

Then we should be able to add other provider implementations of the base class to cover, for example, Google Cloud Firestore and Azure Cosmos DB. Mentioned in https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/1612.

Reply:
gotcha, yes, it makes to the most sense to add it there, ill look at it.